### ISSUE TITLE
Expand Filters and remove unwanted parameters from Handlers

### ISSUE THREAD

**Issue by Eldinnie** [python-telegram-bot/python-telegram-bot#835]
It's been mentioned before, but I wanted a placemarker to keep it. Also see the discussion in #832 
There's a few handlers right now that have the arguments (`message_updates`, `channel_post_updates`, `edited_updates` and the deprecated `allow_edited`) These being `RegexHandler` and `MessageHandler`.

We have a good system in place to decide what update to deliver to a handler with the `Filters` module, and have these extra (legacy) arguments in these handlers is not needed and inconsistent. Also regard Issue #831 .
I propose to:

- Remove the arguments mentioned above from `RegexHandler` and `MessageHandler`
- Add the following filters to the `Filters` module:
  - message_update
  - edited_message_update
  - message_updates (both above)
  - channel_post_update
  - edited_channel_post_update
  - channel_updates (both above)
  - somethingtobenamed (for all above)

The end result will be that either of these update types can be separately filtered for `RegexHandler`, `MessageHandler` and `CommandHandler`.

However we need to decide on a default behavior and how to implement it.I think it could be easily implemented to let all these handlers default to `update.message` if filters list is empty:
```python
if not filters:
  self.filters = Filters.message
```
But whenever a user only specifies a `Filters.group` default behavior will be gone and will fire errors when a messagehandler tries to fire for `update.callback_query` for example.
Love to hear your thoughts.

**Comment by jh0ker** [comment]
I'm wondering if we actually need to continue the `RegexHandler` or if we can integrate it with `MessageHandler`. Or perhaps we can inherit from `MessageHandler`. What do you think?

**Comment by JosXa** [comment]
@jh0ker I've made a couple experiments that corroborate your idea. What I tried was to have a `Filters.regex(pattern)` and it made the handlers a lot cleaner since I could use other filters in conjunction. 
Now the only problem with that is the question of how to pass the `groups` and `groupdict` when there are *multiple* regex filters. One idea would be to require **named** `pass_groups` flags that are passed to the callback in a dict. Another possibility might be to build our own **abstraction** around what was matched by the several regex filters, in that we merge their individual groupdicts together to a bigger one that has all the captured groups. I'm still quite unsure.

On the same note, do we actually still need the `CommandHandler`? *Food for thought.*

@Eldinnie I agree with your proposed changes. The last bullet point, `somethingtobenamed`, are you sure it's needed? I've racked my head trying to come up with a fitting denotation but I've ended up with the conclusion that filtering for both channel posts and normal messages is rarely needed, and it should be done manually by combining the filters imho. Also, a MessageHandler should **never** fire for callback queries, that's a defensive measure we need to build in.
